styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -570,11 +571,28 @@ private void setupMetrics(
           workflow.id(),
           () -> stateManager.getActiveStatesCount(workflow.id()));
 
+      final Optional<Workflow> existingWorkflow = cache.workflow(workflow.id());
+      if (existingWorkflow.isPresent()) {
+        if (!isGreaterOrEqualApiVersion(workflow, existingWorkflow.get())) {
+          return;
```

Comment:
Indeed. Shall we address this later when validating a workflow, e.g. cron syntax? We would need to introduce a new exception and throw it out.

I will anyway put FIXME here.